cpp/src/arrow/util/converter.h

```diff
@@ -52,7 +52,15 @@ class Converter {
     return Init(pool);
   }
 
-  virtual Status Append(InputType value) = 0;
+  virtual Status Append(InputType value) { return Status::NotImplemented("Append"); }
```

Comment:
@bkietz I think we should keep both `Append` and `Extend` since we wrap the converter object with the `Chunker`, so if one implementation (like the python one) choose to use `Append` then we don't need to subclass the chunker class.